theano/misc/tests/test_pkl_utils.py

```diff
@@ -0,0 +1,57 @@
+import numpy
+from numpy.testing import assert_allclose
+from nose.plugins.skip import SkipTest
+
+import theano
+import theano.sandbox.cuda as cuda_ndarray
+
+from theano.sandbox.cuda.type import CudaNdarrayType
+from theano.sandbox.cuda.var import CudaNdarraySharedVariable
+from theano.sandbox.rng_mrg import MRG_RandomStreams
+from theano.misc.pkl_utils import dump, load
+
+
+def test_dump_load():
+    if not cuda_ndarray.cuda_enabled:
+        raise SkipTest('Optional package cuda disabled')
+
+    x = CudaNdarraySharedVariable('x', CudaNdarrayType((1, 1), name='x'),
+                                  [[1]], False)
+
+    with open('test', 'wb') as f:
+        dump(x, f)
+
+    with open('test', 'rb') as f:
+        x = load(f)
+
+    assert x.name == 'x'
+    assert_allclose(x.get_value(), [[1]])
+
+
+def test_dump_load_mrg():
+    if not cuda_ndarray.cuda_enabled:
+        raise SkipTest('Optional package cuda disabled')
```

Comment:
Instead of skipping it, you could have use_cuda=cuda_ndarray.cuda_enabled bellow.